src/doc/tutorials/PythonExpressions.rst

```diff
@@ -0,0 +1,202 @@
+It demonstrates accessing the coordinates and topology of the mesh as well as a variable.
+
+Here is the example script.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 11-48
+
+Let's start off by creating a Pseudocolor plot from the expression.
+
+1. Copy the script into the *Python expression script* editor.
+2. Click the *Apply* button.
+3. Go to *Variables->MyExpression* to change the variable to the expression.
+
+.. figure:: images/PythonExpression-Plot1.png
+
+   The Pseudocolor plot of MyExpression
+
+Now let's take a look at the script and see what each portion does.
+
+The ``__init__`` method provides information about the expression, including
+
+* That it inherits from ``SimplePythonExpression``.
+* The name of the expression.
+* A description of the expression.
+* A flag indicating that the output is not a point centered value.
+* A flag that the output is a scalar.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 13-18
+
+The ``modify_contract`` method can be used to request special information for the expression from VisIt_.
+In this case it is a no-op.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 19-20
+
+The ``derive_variable`` method performs the real work of the expression.
+It is passed the input vtkDataSet and the domain_id.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 21-23
+
+The following lines get the vtkDataArray for the cell values and the number of cells.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 24-25
+
+The following lines set the output vtkDataArray to be an array of floats with 1 component and ncells.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 26-28
+
+Now we loop over the cells, setting the output value for each cell.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 29
+
+The following lines get the current cell and the number of edges in the cell.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 30-31
+
+Now we loop over the edges, calculating the sum of the lengths of the edges.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 32-34
+
+We calculate the length of the edge from the 3D coordinates of the end points of the edge, which we add to the sum.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 35-42
+
+Once we have summed the lengths of the edges we multiply the sum by the cell value and set it in the result.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 43-45
+
+Once we have finished processing all the cells, we return the vtkDataArray.
+
+.. literalinclude:: ../../test/tests/hybrid/python_example_1.vpe
+   :lines: 46
+
+Using VTK in Python
+-------------------
+
+The VTK Python interface mirrors the C++ interface.
```

Comment:
Maybe link to the C++ interface here.

Also, here are some python examples hosted by kitware...

https://kitware.github.io/vtk-examples/site/Python/

Reply:
Oh, ok, you link to specific objects C++ below. Thats fine.